python/lsst/dax/obscore/siav2.py

```diff
@@ -0,0 +1,523 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ["SIAv2Handler", "siav2_query"]
+
+import logging
+import math
+from collections import defaultdict
+from collections.abc import Iterable, Iterator
+from typing import Any, Self
+
+import astropy.io.votable
+import astropy.time
+from lsst.daf.butler import Butler, DimensionGroup, Timespan
+from lsst.daf.butler.pydantic_utils import SerializableRegion, SerializableTime
+from lsst.sphgeom import Region
+from pydantic import BaseModel, field_validator
+
+from .config import ExporterConfig, WhereBind
+from .obscore_exporter import ObscoreExporter
+
+_LOG = logging.getLogger(__name__)
+
+
+class Interval(BaseModel):
+    """Representation of a simple interval."""
+
+    start: float
+    """Start of the interval."""
+    end: float
+    """End of the interval."""
+
+    @classmethod
+    def from_string(cls, string: str) -> Self:
+        """Create interval from string of form 'START END'.
+
+        Parameters
+        ----------
+        string : `str`
+            String representing the interval. +Inf and -Inf are allowed.
+            If there is only one number the start and end interval are the
+            same.
+
+        Returns
+        -------
+        interval : `Interval`
+            The derived interval.
+        """
+        interval = [float(b) for b in string.split()]
+        if len(interval) == 1:
+            interval.append(interval[0])
+        return cls(start=interval[0], end=interval[1])
+
+    def overlaps(self, other: Interval) -> bool:
+        """Return whether two intervals overlap.
+
+        Parameters
+        ----------
+        other : `Interval`
+            Interval to check against.
+
+        Returns
+        -------
+        overlaps : `bool`
+            `True` if this interval overlaps the other.
+        """
+        return self.end >= other.start and other.end >= self.start
+
+    def __iter__(self) -> Iterator[float]:  # type: ignore
+        return iter((self.start, self.end))
+
+
+class SIAv2Parameters(BaseModel):
+    """Parsed versions of SIAv2 parameters."""
+
+    instrument: str | None = None
+    pos: SerializableRegion | None = None
+    time: Timespan | SerializableTime | None = None
+    band: Interval | None = None
+    exptime: Interval | None = None
+    calib: frozenset[int] = frozenset()
+
+    @field_validator("calib")
+    @classmethod
+    def check_calib(cls, calib: frozenset[int]) -> frozenset[int]:
+        valid_calib = {0, 1, 2, 3}
```

Comment:
Hard-coding valid values here is probably not the best way to do it, maybe we need some `obscore_constants` module or something like it?

Reply:
Looks like POL and CALIB have hard-coded values and we aren't going to be using POL. I can make it a frozenset package global for now.